### PR TITLE
[native] Switch C++ standard to C++23

### PR DESCRIPTION
**Pull request by grendello**
Whenever `size_t` or `ssize_t` types are used and an integer literal is
involved, use the `uz` or `z` literal suffix.  This is to make the
intent explicit as well as to avoid hidden type conversions.  Despite
being more verbose, this makes the code more type-safe.

### CODE REVIEW

File: src/native/monodroid/monodroid-tracing.cc [hunk 1]
```diff
@@ -22,7 +22,7 @@ void
 MonodroidRuntime::log_traces (JNIEnv *env, TraceKind kind, const char *first_line) noexcept
 {
 	if (!tracing_init_done) {
-		std::lock_guard lock (tracing_init_lock);
```
Comment (jonpryor) [inline src/native/monodroid/monodroid-tracing.cc:25]: Now I have to wonder why a7b576885624d09c4cc9d65412da2a7558fd3b36 didn't use `xamarin::android::lock_guard` when it was introduced in b90d3ab717.

Comment (grendello) [reply src/native/monodroid/monodroid-tracing.cc:25]: My mistake, I missed it back then.

**Comment by grendello** [comment]
The MAUI integration tests appear to fail in other PRs

**Pull request merged**